shared-bindings/adcbuffer/__init__.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * SPDX-FileCopyrightText: Copyright (c) 2022 Lee Atkinson, MeanStride Technology, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include <stdint.h>
+#include "py/obj.h"
+#include "py/runtime.h"
+#include "shared-bindings/microcontroller/Pin.h"
+#include "shared-bindings/adcbuffer/__init__.h"
+#include "shared-bindings/adcbuffer/BufferedInput.h"
+
+//| """Analog buffered hardware support
+//|
+//| The `adcbuffer` module contains classes to provide access to analog-to-digital
+//| conversion and digital-to-analog (DAC) for multiple value transfer.
+//|
+//| All classes change hardware state and should be deinitialized when they
+//| are no longer needed if the program continues after use. To do so, either
+//| call :py:meth:`!deinit` or use a context manager. See
+//| :ref:`lifetime-and-contextmanagers` for more info.
+//|
+//| For example::
+//|
+//|   import adcbuffer
+//|   import array
+//|   from board import *
+//|
+//|   length = 5000000
+//|   mybuffer = array.array("H", [0] * length)
+//|   adcbuf_obj = adcbuffer.BufferedInput(GP26, mybuffer, length)
+//|   adcbuffer.readmultiple()
+//|   print(*mybuffer)
+//|   adcbuf_obj.deinit()
+//|
+//| This example will initialize the the device, read and fill
+//| :py:data:`~adcbuffer.BufferedInPut` to mybuffer and then
+//| :py:meth:`~adcbuffer.BufferedInPut.deinit` the hardware. The last step is optional
+//| because CircuitPython will do it automatically after the program finishes.
```

Comment:
I'd leave off the "The last step is optional...", because in the usual case, the program is going to run forever, sampling, say every few seconds, or on command, or whatever. You could then just leave out the `.deinit()` line in the example.

Reply:
Changed as suggested.